### ISSUE TITLE
Significant Increase in Loss when Adjusting traj_per_sample Parameter

### ISSUE THREAD

**Issue by sinkers-lan** [facebookresearch/co-tracker#97]
I am currently performing full fine-tuning. When I attempt to adjust the `traj_per_sample` parameter from 768 to 384 during training, the average loss approximately doubles. When I adjust the `traj_per_sample` parameter from 768 to 256, the average loss increases by about three times. 

After observing this phenomenon, I carefully reviewed the code for the loss function and noticed that the loss is divided by N at the end:
```python
total_balanced_loss += balanced_loss / float(N)
```
This line of code can be found [here](https://github.com/facebookresearch/co-tracker/blob/9ed05317b794cd177674e681321780614a65e073/cotracker/models/core/cotracker/losses.py#L37).

Similarly,
```python
total_flow_loss += flow_loss / float(N)
```
This line of code can be found [here](https://github.com/facebookresearch/co-tracker/blob/9ed05317b794cd177674e681321780614a65e073/cotracker/models/core/cotracker/losses.py#L60).

I believe this might be the cause of the aforementioned increase in loss. This is because before dividing the loss by N, the `reduce_masked_mean` function already computes the mean across various dimensions. Dividing by N again leads to a larger loss when N is smaller.

I think this might be a logical error in the code. Your guidance on this issue would be greatly appreciated.

**Comment by nikitakaraevv** [comment]
Hi @sinkers-lan, thank you for catching this! This is indeed a logical error. We'll fix it in the next version that's coming up.

**Comment by G1tQuality** [comment]
Hi @sinkers-lan, does full fine-tuning mean that training on new datasets which is on other field? I read your comments under another issue about prepare kubric training datasets. Could you please give me some advice about full fine-tuning for CoTraker? If I wanna apply CoTracker on my own datasets, is it necessary for me to train it on kubric? Or just start "full fine-tuning"?  I apologize for causing you any inconvenience as a noob in deep learning.
